### ISSUE TITLE
owners remove/ add response does not fit documentation

### ISSUE THREAD

**Issue by 2ndDerivative** [rust-lang/cargo#12056]
### Problem

The Registry Web API documentation says the Owner --remove command needs a JSON object
```json
{
    "ok":true
}
```
even though it needs the same `OwnerResponse` object as the add command, which also has a `msg` field.
This causes registries that fit the documentation API to error out the user for an incomplete JSON when removing a user.

### Steps

Respond to the Owner remove API request with a JSON object without the "msg" field.
Cargo will error out with 
```
error: failed to remove owners from crate `[crate-name]` on registry at [endpoint]

Caused by:
  missing field `msg` at line 1 column 11
 ```

### Possible Solution(s)

Either the OwnerResponse object should probably be split into AddResponse and RemoveResponse to fit the documentation, or the msg field on the remove method should be used, or the msg field should be made optional.

**Comment by weihanglo** [comment]
Just looked into [crates.io implementation][impl], it always responds a JSON of `{ ok, msg }`. 

Given Cargo already fails when missing that field, I would recommend adding the `msg` field to the documentation. It shouldn't be a breaking change to registry implementors.

[impl]: https://github.com/rust-lang/crates.io/blob/fea6338cfa59f13ecfc06b685cec67aa926d1edb/src/controllers/krate/owners.rs#L57-L73

**Comment by weihanglo** [comment]
Place that needs a tweak:

https://github.com/rust-lang/cargo/blob/a285008c8ec3c19896740e153332a2bd1ef93086/src/doc/src/reference/registry-web-api.md?plain=1#L307-L312

**Comment by 2ndDerivative** [comment]
I'd also suggest putting in the message display into cargo, I can do that later

**Comment by Bhardwaj-Himanshu** [comment]
Hi everyone, from my understanding 
this-
 ```javascript 
 { 
     // Indicates the remove succeeded, always true. 
     "ok": true 
 } 
 ``` 
 should be replaced to something like-
  ```javascript 
 { 
     // some kind of message here
     "ok": true 
 } 
 ``` 
 
 But being unsure of how `AddResponse and RemoveResponse ` work, I am unable to sense what changes to make and how.
 Are these 2 seperate functions which need to be added as-
 ```js
Addresponse { 
     // something_1
     "ok": true 
 } 
 ```
 and 
 ```js
 RemoveResponse{ 
     // something_1
     "ok": true 
 }
 ```
 
 Or Add response and Remove response are just messages that need to be displayed whenever user interacts with this?
 
 Also, let me know if someone is upto some ideas and how he/she is working on this.
 Thanks

**Comment by 2ndDerivative** [comment]
No  the Json object response to `cargo owner --remove` needs to be replaced by 
```
{
    "message": "Some message here,
    "ok": true
}
```

because the reponse cargo expects is the same as the response to `cargo owner --add`.

**Comment by Bhardwaj-Himanshu** [comment]
So the message which needs to go in there should be like

```js
message: "task performed successfully"

#not writing owner removed or owner added here, as mentioned by you, the object responds similarly for both the cases.

```

**Comment by 2ndDerivative** [comment]
The message field in the reponse to `owner --remove` is not used in any way, but cargo expects it to just exist. So so far there can be anything there.

what we need to change is the documentation to recommend putting a message there to API implementors and warning that it's needed but not used (yet).

**Comment by hi-rustin** [comment]
@weihanglo I guess we can close this issue now.

**Comment by weihanglo** [comment]
Surely. Thanks!

Fixed by #12068
